sdks/java/harness/src/main/java/org/apache/beam/fn/harness/state/FnApiStateAccessor.java

```diff
@@ -1086,6 +1225,45 @@ private <T> OrderedListUserState<T> createOrderedListUserState(
     return rval;
   }
 
+  public static class OrderedListStateKey implements Weighted {
+    private final String pTransformId;
+    private final String stateId;
+    private final ByteString window;
+    private final ByteString key;
+    private final int hash;
+
+    public OrderedListStateKey(StateKey.OrderedListUserState proto) {
+      this.window = proto.getWindow();
+      this.key = proto.getKey();
+      this.pTransformId = proto.getTransformId();
+      this.stateId = proto.getUserStateId();
+      this.hash = Objects.hash(OrderedListStateKey.class, window, key, pTransformId, stateId);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (!(o instanceof FnApiStateAccessor.OrderedListStateKey)) {
+        return false;
+      }
+      OrderedListStateKey other = (OrderedListStateKey) o;
+      return hash == other.hash
+          && pTransformId.equals(other.pTransformId)
+          && stateId.equals(other.stateId)
+          && window.equals(other.window)
+          && key.equals(other.key);
+    }
+
+    @Override
+    public int hashCode() {
+      return hash;
+    }
+
+    @Override
+    public long getWeight() {
```

Comment:
These three Weighted implementations look identical. Any consideration to do so or consider merge them?